src/care/PluginData.h

```diff
@@ -0,0 +1,65 @@
+#ifndef _CARE_PluginData_H_
+#define _CARE_PluginData_H_
+
+#include "chai/PointerRecord.hpp"
+#include "care/config.h"
+#include <vector>
+#include <functional>
+#include <unordered_map>
+
+namespace chai {
+   struct PointerRecord;
+}
+
+using ActionMap = std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>>;
+
+namespace care{
+   //class for shared plugin functions and variables
+   class PluginData	{
+      public:
+         PluginData() = default;
+
+         static void setFileName(const char * name);
+
+         static void setLineNumber(int num);	
+
+         static const char * getFileName();
+         
+         static int getLineNumber();   
+
+         static void setParallelContext(bool isParallel);
+
+         static bool isParallelContext();
+
+         static bool post_parallel_forall_action_registered(void * key); 
+
+         static ActionMap get_post_parallel_forall_actions();
+
+         static void register_post_parallel_forall_action(void * key, std::function<void(chai::ExecutionSpace, const char *, int)> action);
+
+         static void clear_post_parallel_forall_actions();
+
+         static std::vector<const chai::PointerRecord*> getActivePointers();
+
+         static void addActivePointer(const chai::PointerRecord* record);
+
+         static void removeActivePointer(const chai::PointerRecord* record);
+
+         static void clearActivePointers();
+
+         static int s_threadID;
+
+      private:
+         static const char * s_file_name;
+
+         static int s_line_number;
+
+         static bool s_parallel_context;
+
+         static std::unordered_map<void *, std::function<void(chai::ExecutionSpace, const char *, int)>> s_post_parallel_forall_actions;
```

Comment:
Sorry one more comment :D . Is this also an ActionMap, or is the signature a bit different?

Reply:
yes you're right, i missed that!

Reply:
No worries! Thanks for making the fix!